runtime/reference/security_model.md

```diff
@@ -0,0 +1,184 @@
+---
+title: "Security Model"
+---
+
+This document outlines the security model of the Deno runtime. Deno is designed
+to be secure by default runtime, however to write secure applications it is
+important to understand the security model of Deno.
+
+Deno's security model is based on the following key principles:
+
+- **No access to I/O by default**: Code executing in a Deno runtime has no
+  access to read or write arbitrary files on the file system, to make network
+  requests or open network listeners, to access environment variables, or to
+  spawn subprocesses.
+- **No limits on the execution of code at the same privilege level**: Deno
+  allows the execution of any code (JS/TS/Wasm) via multiple means, including
+  `eval`, `new Function`, dynamic imports and web workers at the same privilege
+  level with little restriction as to where the code originates (network, npm,
+  JSR, etc).
+- **Multiple invocations of the same application can share data**: Deno provides
+  a mechanism for multiple invocations of the same application to share data,
+  through built in caching and KV storage APIs. Different applications can not
+  see each other's data.
+- **All code executing on the same thread shares the same privilege level**: All
+  code executing on the same thread shares the same privilege level. It is not
+  possible for different modules to have different privilege levels within the
+  same thread.
+- **Code can not escalate it's privileges without user consent**: Code executing
+  in a Deno runtime can not escalate it's privileges without the user agreeing
+  explicitly to an escalation.
+- **The initial static module graph can import local files without
+  restrictions**: All files that are imported in the initial static module graph
+  can be imported without restrictions, so even if a explicit read permission is
+  not granted for that file. This does not apply to any dynamic module imports.
+
+These key principles are designed to provide an environment where a user can
+execute code with minimal risk of harm to the host machine or network. The
+security model is designed to be simple to understand and to provide a clear
+separation of concerns between the runtime and the code executing within it. The
+security model is enforced by the Deno runtime, and is not dependent on the
+underlying operating system.
+
+## Permissions
+
+By default, access to most system I/O is denied. There are some I/O operations
+that are allowed in a limited capacity, even by default. These are described
+below.
+
+To enable these operations, the user must explicitly grant permission to the
+Deno runtime. This is done by passing the `--allow-read`, `--allow-write`,
+`--allow-net`, `--allow-env`, and `--allow-run` flags to the `deno` command.
```

Comment:
I was about to suggest you link to https://docs.deno.com/runtime/fundamentals/security/ but maybe we should roll this document into that already existing document?

Reply:
seconded